packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -132,11 +190,37 @@ export class StateMachine extends StateMachineBase {
 
         this.stateMachineType = props.stateMachineType ? props.stateMachineType : StateMachineType.STANDARD;
 
+        let loggingConfiguration: CfnStateMachine.LoggingConfigurationProperty | undefined;
+        if (props.loggingConfiguration && props.loggingConfiguration.destinations.length > 0) {
+            const conf = props.loggingConfiguration;
+            loggingConfiguration = {
+                destinations: conf.destinations.map(loggroup => ({ cloudWatchLogsLogGroup: { logGroupArn: loggroup.logGroupArn } })),
+                includeExecutionData: conf.includeExecutionData,
+                level: conf.level || "ERROR"
+            };
+            // https://docs.aws.amazon.com/step-functions/latest/dg/cw-logs.html#cloudwatch-iam-policy
+            this.addToRolePolicy(new iam.PolicyStatement({
+              effect: iam.Effect.ALLOW,
+              actions: [
+                "logs:CreateLogDelivery",
+                "logs:GetLogDelivery",
+                "logs:UpdateLogDelivery",
+                "logs:DeleteLogDelivery",
+                "logs:ListLogDeliveries",
+                "logs:PutResourcePolicy",
+                "logs:DescribeResourcePolicies",
+                "logs:DescribeLogGroups"
+              ],
+              resources: ["*"]
+            }));
```

Comment:
We try our best to keep least privilege, so avoid `*` resource.

Use `loggroup.grant()` API instead.

Reply:
Those permissions are not bound to resources but more general CloudWatch Logs permissions allowing StepFunctions to write into the account: https://docs.aws.amazon.com/IAM/latest/UserGuide/list_amazoncloudwatchlogs.html

There is no resource which can be referenced.

Reply:
As I mentioned the permissions are not bound to a resource but for the whole account. That's also shown in the docs I linked. It does not work with a LogGroup/LogStream restriction.